uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -385,32 +425,37 @@ define([
       function loadAbsenceTypes() {
         // Fetch all the absence types, except for the sickness ones
         return AbsenceType.all({
-            is_sickness: false
+            is_sick: false
           })
           .then(function (absenceTypes) {
             var absenceTypesIds = absenceTypes.map(function (absenceType) {
               return absenceType.id;
             });
 
-            return {
+            absenceTypesAndIds = {
               types: absenceTypes,
               ids: absenceTypesIds
             };
+
+            return absenceTypesAndIds;
           })
-          .then(loadAbsenceTypesFromEntitlements);
+          .then(setAbsenceTypesFromEntitlements);
```

Comment:
I think sequence of then's keep the code clean and related code is kept in one block